main.go

```diff
@@ -62,18 +62,14 @@ func main() {
 			must(fmt.Errorf("%s not in ~/.aws/roles", role))
 		}
 
-		if os.Getenv("ASSUMED_ROLE") != "" {
-			// Clear out any previously set AWS_ environment variables so
-			// they aren't used by this call
-			cleanEnv()
-		}
+		// Clear out any previously set AWS_ environment variables so
+		// they aren't used by this call
+		cleanEnv()
```

Comment:
I think `cleanEnv()` should actually just get removed. If we don't want `aws-sdk-go` to use creds from the environment, then we should just swap out the credential chain with an explicit `SharedCredentialsProvider`. I think it'd look something like:

```golang
config := aws.Config{Credentials: &credentials.SharedCredentialsProvider{Filename: "", Profile: ""}}
session := session.New(config)
```

`cleanEnv()` is some cruft from when this tool would shell out to the `aws` cli.